pallets/parachain-staking/src/lib.rs

```diff
@@ -2790,78 +2924,22 @@ pub mod pallet {
 	where
 		T: Config + pallet_authorship::Config + pallet_session::Config,
 	{
-		/// Compute coinbase rewards for block production and distribute it to
-		/// collator's (block producer) and its delegators according to their
-		/// stake and the current InflationInfo.
-		///
-		/// The rewards are split between collators and delegators with
-		/// different reward rates and maximum staking rates. The latter is
-		/// required to have at most our targeted inflation because rewards are
-		/// minted. Rewards are immediately available without any restrictions
-		/// after minting.
-		///
-		/// If the current staking rate is below the maximum, each collator and
-		/// delegator receives the corresponding `reward_rate * stake /
-		/// blocks_per_year`. Since a collator can only author blocks every
-		/// `MaxSelectedCandidates` many rounds, we multiply the reward with
-		/// this number. As a result, a collator who has been in the set of
-		/// selected candidates, eventually receives `reward_rate * stake` after
-		/// one year.
-		///
-		/// However, if the current staking rate exceeds the max staking rate,
-		/// the reward will be reduced by `max_rate / current_rate`. E.g., if
-		/// the current rate is at 50% and the max rate at 40%, the reward is
-		/// reduced by 20%.
-		///
-		/// # <weight>
-		/// Weight: O(D) where D is the number of delegators of this collator
-		/// block author bounded by `MaxDelegatorsPerCollator`.
-		/// - Reads: CandidatePool, TotalCollatorStake, Balance,
-		///   InflationConfig, MaxSelectedCandidates, Validators,
-		///   DisabledValidators
-		/// - Writes: (D + 1) * Balance
-		/// # </weight>
+		// TODO: Docs
+		/// Increments the reward counter of the block author by the number of
+		/// collators.
 		fn note_author(author: T::AccountId) {
-			let mut reads = Weight::one();
-			let mut writes = Weight::zero();
 			// should always include state except if the collator has been forcedly removed
 			// via `force_remove_candidate` in the current or previous round
-			if let Some(state) = CandidatePool::<T>::get(author.clone()) {
-				let total_issuance = T::Currency::total_issuance();
-				let TotalStake {
-					collators: total_collators,
-					delegators: total_delegators,
-				} = <TotalCollatorStake<T>>::get();
-				let c_staking_rate = Perquintill::from_rational(total_collators, total_issuance);
-				let d_staking_rate = Perquintill::from_rational(total_delegators, total_issuance);
-				let inflation_config = <InflationConfig<T>>::get();
+			if CandidatePool::<T>::get(author.clone()).is_some() {
```

Comment:
Probably unrelated, but we can use references to access the storage instead of cloning the key. Like `CandidatePool::<T>::get(&author)`.